engine/src/main/java/org/hibernate/validator/internal/engine/ValidationContext.java

```diff
@@ -659,22 +661,17 @@ private BeanPathMetaConstraintProcessedUnit(Object bean, Path path, MetaConstrai
 
 		@Override
 		public boolean equals(Object o) {
-			if ( this == o ) {
-				return true;
-			}
```

Comment:
Same here.

Reply:
I've checked. It don't really make a difference, results are almost identical. So I've added these back.